src/client/Client.cc

```diff
@@ -6690,7 +6693,7 @@ void Client::tick()
       check_caps(in, CHECK_CAPS_NODELAY);
   }
 
-  if (!mount_aborted)
+  if (!mount_aborted && _collect_and_send_global_metrics)
```

Comment:
how about directly read from `cct->_conf.get_val<bool>("client_collect_and_send_global_metrics")` instead? I think that would work for the user updated values too. Try that!

Reply:
Yeah, that works, but it's inefficient in the `tick()` thread. I see there have some commits in mds/ were fixing this to avoid get the values directly from the config in this case.